src/schema/python/test_SchemaMigMgr.py

```diff
@@ -0,0 +1,243 @@
+# This file is part of qserv.
+#
+# Developed for the LSST Data Management System.
+# This product includes software developed by the LSST Project
+# (https://www.lsst.org).
+# See the COPYRIGHT file at the top-level directory of this distribution
+# for details of code ownership.
+#
+# This program is free software: you can redistribute it and/or modify
+# it under the terms of the GNU General Public License as published by
+# the Free Software Foundation, either version 3 of the License, or
+# (at your option) any later version.
+#
+# This program is distributed in the hope that it will be useful,
+# but WITHOUT ANY WARRANTY; without even the implied warranty of
+# MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+# GNU General Public License for more details.
+#
+# You should have received a copy of the GNU General Public License
+# along with this program.  If not, see <https://www.gnu.org/licenses/>.
+
+
+import os
+from tempfile import TemporaryDirectory
+import unittest
+from unittest.mock import MagicMock
+
+from schemaMigMgr import Migration, SchemaMigMgr, Uninitialized
+
+
+def makeFiles(directory, *names):
+    for name in names:
+        with open(os.path.join(directory, name), "w") as f:
+            f.write("foo")
+
+
+class SchemaMigMgrTestCase(unittest.TestCase):
+    def test_find_scripts(self):
+        """Verify find_scripts uses the regex function to match expected
+        filename strings and not unexpected ones."""
+
+        class TestMigMgr(SchemaMigMgr):
+            def current_version(self):
+                pass
+
+            def apply_migrations(self, migrations):
+                pass
+
+        self.assertEqual(TestMigMgr.script_match("migrate-0-to-1.sql"), (0, 1))
+        self.assertEqual(TestMigMgr.script_match("migrate-foo-to-1.sql"), None)
+        self.assertEqual(
+            TestMigMgr.script_match("migrate-None-to-4.sql"), (Uninitialized, 4)
+        )
+        self.assertEqual(
+            TestMigMgr.script_match("migrate-None-to-4-somedescriptor.sql"),
+            (Uninitialized, 4),
+        )
+
+    def test_scripts(self):
+        """Verify the default init creates the scripts member variable
+        correctly from migration scripts found in the passed-in directory path.
+        """
+
+        class TestMigMgr(SchemaMigMgr):
+            def current_version(self):
+                pass
+
+            def apply_migrations(self, migrations):
+                pass
+
+        noneToTwo = "migrate-None-to-2.sql"
+        oneToTwo = "migrate-1-to-2.sql"
+
+        with TemporaryDirectory() as tmpDir:
+            makeFiles(tmpDir, noneToTwo, oneToTwo, "unused.sql")
+            migMgr = TestMigMgr(tmpDir)
+            self.assertEqual(
```

Comment:
With `assertCountEqual` you would not need `set()`.

Reply:
that would verify that it found 2, but would not compare the ones that were found to make sure what was found is what was expected, which seems more complete?

Reply:
`assertCountEqual` name is probably confusing, but it compares that elements are equal in both sequences and also that counts of elements are equal too (in case elements appear more than once).